radix-engine/src/kernel/call_frame.rs

```diff
@@ -744,7 +744,7 @@ impl<C, L: Clone> CallFrame<C, L> {
         heap: &'f Heap,
         store: &'f mut S,
         lock_handle: LockHandle,
-    ) -> Result<&'f IndexedScryptoValue, ReadSubstateError> {
+    ) -> Result<(&'f IndexedScryptoValue, bool), ReadSubstateError> {
```

Comment:
Instead of returning `bool` can we pass the callback into this method instead, so that we don't leak weird abstractions in the interface.

Reply:
I added `CallFrameEventHandler` as an parameter through which information about reading from heap is passed back to the Kernel (to `KernelHandler`).